src/ansi-c/scanner.l

```diff
@@ -928,7 +917,7 @@ xor_eq              { return cpp98_keyword(TOK_XORASSIGN); }
 __decltype          { if(PARSER.cpp98 &&
                          (PARSER.mode==configt::ansi_ct::flavourt::GCC ||
                           PARSER.mode==configt::ansi_ct::flavourt::CLANG))
-                        return cpp98_keyword(TOK_DECLTYPE);
+                        return conditional_keyword(PARSER.cpp98, TOK_DECLTYPE);
                       else
                         return make_identifier();
                     }
```

Comment:
I'd suggest that (here and also in a few other places) we now combine all of this logic:
````c++
__decltype          { return conditional_keyword(
                                PARSER.cpp98 &&
                                (PARSER.mode==configt::ansi_ct::flavourt::GCC ||
                                PARSER.mode==configt::ansi_ct::flavourt::CLANG),
                                TOK_DECLTYPE);
                    }
````
(indent perhaps to be adjusted)

Reply:
done